cmd/orb.go

```diff
@@ -740,3 +758,144 @@ https://circleci.com/orbs/registry/orb/%s
 
 	return nil
 }
+
+type OrbSchema struct {
+	Version     string                   `yaml:"version,omitempty"`
+	Description string                   `yaml:"description,omitempty"`
+	Display     interface{}              `yaml:"display,omitempty"`
+	Orbs        interface{}              `yaml:"orbs,omitempty"`
+	Commands    interface{}              `yaml:"commands,omitempty"`
+	Executors   interface{}              `yaml:"executors,omitempty"`
+	Jobs        interface{}              `yaml:"jobs,omitempty"`
+	Examples    map[string]ExampleSchema `yaml:"examples,omitempty"`
+}
+
+type ExampleUsageSchema struct {
+	Version   string      `yaml:"version,omitempty"`
+	Orbs      interface{} `yaml:"orbs,omitempty"`
+	Jobs      interface{} `yaml:"jobs,omitempty"`
+	Workflows interface{} `yaml:"workflows"`
+}
+
+type ExampleSchema struct {
+	Description string             `yaml:"description,omitempty"`
+	Usage       ExampleUsageSchema `yaml:"usage,omitempty"`
+	Result      ExampleUsageSchema `yaml:"result,omitempty"`
+}
+
+func packOrb(opts orbOptions) error {
+	// Travel our Orb and build a tree from the YAML files.
+	// Non-YAML files will be ignored here.
+	_, err := os.Stat(filepath.Join(opts.args[0], "@orb.yml"))
+	if err != nil {
+		return errors.Wrap(err, "@orb.yml file not found, are you sure this is the Orb root?")
+	}
+
+	tree, err := filetree.NewTree(opts.args[0], "executors", "jobs", "commands", "examples")
+	if err != nil {
+		return errors.Wrap(err, "An error occurred trying to build the tree")
```

Comment:
Many of these error messages seem confusing to the user:
```
Error: @orb.yml file not found, are you sure this is the Orb root?: stat @orb.yml: no such file or directory
Error: An error occurred trying to build the tree: ...
Error: Failed trying to travel Orb YAML: ...
```

The colon separation is one issue (maybe we should look at replacing that with newlines at some point but probably out of scope here), making errors that end with punctuation look bad.

I'd also say these seem more directed towards the programmer (saying where in the code things failed) than at the user, who might not understand something like "what does it mean to travel Orb YAML" or "what tree is orb pack building?" I think an error message like "an unexpected error occurred" might be better (still wrapping the underlying error) because it removes information that isn't relevant to the user.